### PR TITLE
LLAPI refactor - Properties.

### PR DESCRIPTION
**Pull request by RubyNova**
This PR contains all our upcoming changes to our C++ properties, as discussed per the code review.

### CODE REVIEW

File: include/NovelRT/Animation/SpriteAnimatorFrame.h [hunk 1]
```diff
@@ -27,12 +27,12 @@ namespace NovelRT::Animation {
       _texture = value;
     }
 
-    inline float getDuration() const noexcept {
+    inline const float& duration() const noexcept {
```
Comment (tannergooding) [inline include/NovelRT/Animation/SpriteAnimatorFrame.h]: Do we have an issue tracking changing this to `Timestamp`?

Comment (RubyNova) [reply include/NovelRT/Animation/SpriteAnimatorFrame.h]: No but it is on my to do from the review and will be doing it in a separate PR. 😄

File: include/NovelRT/Input/InteractionObject.h [hunk 2]
```diff
@@ -23,9 +23,15 @@ namespace NovelRT::Input {
...
     void executeObjectBehaviour() final;
     virtual bool validateInteractionPerimeter(const Maths::GeoVector<float>& mousePosition) const = 0;
-    KeyCode getSubscribedKey() const;
-    void setSubscribedKey(KeyCode key);
+
+    inline KeyCode& subscribedKey() noexcept {
+      return _subscribedKey;
+    }
+
+    inline const KeyCode& subscribedKey() const noexcept {
```
Comment (tannergooding) [inline include/NovelRT/Input/InteractionObject.h]: nit: We've ordered the constant version first everywhere else so far.

**Comment by RubyNova** [comment]
Oh, so I was correct in not touching the other properties?

**Pull request merged**